### PR TITLE
Variable injection in feature files (scenario variables only)

### PR DESCRIPTION
**Pull request by seantdg**

### CODE REVIEW

**Comment by seantdg** [comment]
@seymen @sauliuz let me know what you think.

**Comment by Mattzr** [comment]
+1, could be very useful.
I was actually working on something similar with the ability to inject variables from yaml/json file. In that case variables could be dynamically injected into the scenario context. Happy to accommodate with this PR.

@seantdg @seymen @sauliuz 
What do you think ?

File: source/test/features/step_definitions/apickli-test.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ module.exports = function() {
 		this.apickli = new apickli.Apickli('http', 'httpbin.org');
 		callback();
 	});
+    
+    this.Given(/^I store the value (.*) as (.*) in scenario scope$/, function (value, variable, callback) {
```
Comment (seymen) [inline source/test/features/step_definitions/apickli-test.js]: This is breaking tests as cucumber confuses it with
```
/^I store the value of response header (.*) as (.*) in scenario scope$/
```

**Comment by seymen** [comment]
@seantdg - I think it is useful to expose replaceVariables out to other step definitions in case a more 'business-friendly' step def want to reuse the functionality.

I couldn't understand the reason for separating `replaceVariables` and `replaceScopeVariables`. 

Need to keep positive tests under @core tag so gulp can execute them by default.

I will do a commit to fix those.

**Pull request merged**